mysql-test/lib/mtr_report.pm

```diff
@@ -297,24 +299,24 @@ sub mtr_report_stats ($$$$) {
 
   foreach my $tinfo (@$tests)
   {
-    if ( $tinfo->{failures} )
+    if ( $tinfo->{'result'} eq MTR_RES_PASSED )
```

Comment:
Here is just switch of the logic. If possible please leave in comment the proof why this is needed. Proof consists of invoking of the test as well as the outputs (xml-report) before this change and after this change. I don't see what could be different. `test{failures}` are set only if there is a valgrind report and test is failed.

Reply:
This is necessary for any test that will be repeated. If the test failed once, it will be marked as failed, even if it succeeds in the next restart. If there is a failed test in this restart, the failures counter will be at least equal to one. Accordingly, it is necessary to check the result.

Reply:
The result and failures are not the same thing. failures - the number of failures, that is, if we restart the test that fell before but now is success, then it will have the MTR_RES_PASSED status, but failures = 1. If we first check the result, then this test will fall into the successful ones.

Reply:
- There is no failure if there is no `MTR_RES_FAILED` of the result, meaning the test is failed, so there are the same thing.
- No it will not happen like you mentioned, function `mtr_report_test_passed ` will be called (will be executed) in case when everything (even repeating of test ) related to the test finishes, see my previous comment please and code . That means in that case `test->{failures}` for example will be 2,3,4... number of repeated times of execution of the test.

Reply:
These changes are relevant only for tests with retry indication and going to the end and not being interrupted on the first failed test. Script:

1. Suppose one of the tests failed. 
2. This test will restart.
3. The test was successful again.

In this case, there will be two information about the test (since we now add all information about each test, including restarts). Logically, one of the tests will succeed, the other will fail. Аt the same time, both of them will have `test->{failures} = 1`. But at the same time, their statuses will be different (`MTR_RES_PASSED`, `MTR_RES_FAILED`). Accordingly, if first of all look at the status, then one of the tests will be credited as successful, the other has fallen. Otherwise, both of them will be credited to the failed (illogical).